src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/InstalledAndTransitivePackageFeed.cs

```diff
@@ -28,12 +28,17 @@ public override async Task<SearchResult<IPackageSearchMetadata>> ContinueSearchA
             var searchToken = continuationToken as FeedSearchContinuationToken ?? throw new InvalidOperationException(Strings.Exception_InvalidContinuationToken);
             cancellationToken.ThrowIfCancellationRequested();
 
+            PackageCollectionItem[] matchesInstalled = PerformLookup(_installedPackages.GetLatestCore(), searchToken);
+
             // Remove transitive packages from project references
-            IEnumerable<PackageCollectionItem> pkgsWithOrigins = _transitivePackages
-                .Where(t => t.PackageReferences.Any(x => x is ITransitivePackageReferenceContextInfo y && y.TransitiveOrigins.Any()));
+            IEnumerable<PackageCollectionItem> transitivePkgsWithOrigins = _transitivePackages
+                .Where(transitivePkg => transitivePkg
+                    .PackageReferences
+                    .All(pkgRef => pkgRef is ITransitivePackageReferenceContextInfo trPkgRef && trPkgRef.TransitiveOrigins.Any()));
+            PackageCollectionItem[] matchesTransitive = PerformLookup(transitivePkgsWithOrigins.GetLatestCore(), searchToken);
```

Comment:
No need to cache this in a variable since it's only ever used once.

Reply:
Removed 'caching' variables.